File: api/src/main/java/io/bunnies/baas/resources/BunnyResourcesSingleton.java
package io.bunnies.baas.resources;

import io.bunnies.baas.metadata.DerivedMetadata;
import io.bunnies.baas.metadata.SpecifiedMetadata;

import java.util.Set;

public class BunnyResourcesSingleton {
    private static BunnyResources bunnyResources;

    public static BunnyResources getInstance(String mediaBaseUrl, DerivedMetadata derivedMetadata, SpecifiedMetadata specifiedMetadata) {
        if (bunnyResources == null) {
            bunnyResources = new BunnyResources(mediaBaseUrl, derivedMetadata, specifiedMetadata);
        }

        return bunnyResources;
    }
}

File: api/src/main/java/io/bunnies/baas/resources/BunnyResourcesSingleton.kt
package io.bunnies.baas.resources

import io.bunnies.baas.metadata.DerivedMetadata
import io.bunnies.baas.metadata.SpecifiedMetadata

object BunnyResourcesSingleton {
    private var bunnyResources: BunnyResources? = null

    fun getInstance(mediaBaseUrl: String, derivedMetadata: DerivedMetadata, specifiedMetadata: SpecifiedMetadata): BunnyResources {
        return bunnyResources ?: createInstance(mediaBaseUrl, derivedMetadata, specifiedMetadata)
    }

    private fun createInstance(mediaBaseUrl: String, derivedMetadata: DerivedMetadata, specifiedMetadata: SpecifiedMetadata): BunnyResources {
        val bunnyResources = BunnyResources(mediaBaseUrl, derivedMetadata, specifiedMetadata)
        this.bunnyResources = bunnyResources

        return bunnyResources
    }
}
